src/util/validation.js

```diff
@@ -0,0 +1,42 @@
+import { CONFIG } from '../constant/config.js';
+import { ERROR_MESSAGE } from '../constant/message.js';
+
+function validateCarNames(carNames) {
+  carNames.forEach((name) => {
+    if (
+      name.length < CONFIG.MIN_CAR_NAME_LENGTH ||
+      name.length > CONFIG.MAX_CAR_NAME_LENGTH
+    ) {
+      throw new Error(ERROR_MESSAGE.INVALID_CAR_NAME);
+    }
+  });
+}
+
+function validateMinimumCarCount(carNames) {
+  if (carNames.length < CONFIG.MIN_CAR_LIST_LENGTH) {
+    throw new Error(ERROR_MESSAGE.MIN_CARS_REQUIRED);
+  }
+}
+
+function validateDuplicateCarNames(carNames) {
+  const nameSet = new Set();
+
+  carNames.forEach((name) => {
+    if (nameSet.has(name)) {
+      throw new Error(ERROR_MESSAGE.CAR_ALREADY_EXISTS);
+    }
+    nameSet.add(name);
+  });
```

Comment:
`carNames`를 순회하는 것은 `O(N)`의 시간 복잡도를 가져요, 그렇다면 `carNames`를 기반으로 한 `Set`을 만드신다면 로직을 더 간단하게 나타낼 수 있을 것 같아요 (배열을 통해서 `Set`을 만드는 것도 `O(N)`의 시간이 걸려요)

그러니 아래와 같이 간단하게 작성하는 방법은 어떠실까요?

```suggestion
  if (new Set(carNames).size !== carNames.length) {
     throw new Error(ERROR_MESSAGE.CAR_ALREADY_EXISTS);
  }
```

Reply:
좋은 방법인 것 같아요! 하나 배워갑니다 ! 👍